### ISSUE TITLE
AIO v7.12.1 after daily backup & auto update, it fails to restart containers (when internet access is temporarily down)

### ISSUE THREAD

**Issue by sunnyd24** [nextcloud/all-in-one#4250]
### Steps to reproduce
1. Lose internet connection temporarily (for whatever reason, internet provider outage or host system network problems)
2. Daily backup starts Sun-Fri (+ update containers on Sat)
3. Daily backup finishes
4. Restarts all containers
5. Fails to start Apache, Nextcloud, Notify Push, Imaginary and Docket Socket Proxy containers.

### Expected behavior 
1. Lose internet connection temporarily (for whatever reason, internet provider outage or host system network problems)
2. Daily backup starts Sun-Fri (+ update containers on Sat)
3. Daily backup finishes
4. Restarts all containers
5. All containers are "running" with no "stopped" containers.
6. If any containers fail to go to "running" state, stop all containers, (start customisable timer **OR** continuously check for internet access) **AND** (queue email/Nextcloud notification to administrator of failure **AND** send notification through some method like PushOver/PushBullet/Telegram/WebSocket).
7. At the end of timer **OR** if internet access is re-established, then attempt to restart container.
8. Send success email/Nextcloud notification to administrator **AND** send notification through some method like PushOver/PushBullet/Telegram/WebSocket.

### Actual behavior 
1. Daily backup starts.
2. Loses internet access temporarily.
3. Containers **Apache, Nextcloud, Notify Push, Imaginary and Docket Socket Proxy** have "stopped", all others are "running", see screenshot below.
4. No notification (queued email or otherwise) of Nextcloud AIO in a failed stuck state and requires manual intervention from administrator to investigate.

### Host OS 
Windows running Hyper VM with Ubuntu 22.04.3 LTS

#### Nextcloud AIO version 
10.12.1

#### Current channel 
latest

#### Other valuable info 

When going through borg daily backup (and update process), several containers failed to restart automatically:

![image](https://github.com/nextcloud/all-in-one/assets/46629122/e5c9ea61-3859-4489-ba64-4137d99737cb)

I suspect it was due to the VM not being able to talk to the internet due to stuck host processes slowing down the host network comms.

May I suggest a new feature is required (to allow master container to attempt restarts automatically at specified intervals for any "stopped" containers if not successfully "running" after backup is complete **OR** wait for internet connection to resolve) **AND** (queue email to administrator if this is triggered AND send notification through some method like PushOver/PushBullet/Telegram/WebSocket))?

See below for master container log:

```
Daily backup script has started
grep: write error: Broken pipe
Connection to nextcloud-aio-apache (172.18.0.13) 443 port [tcp/https] succeeded!
Starting mastercontainer update...
(The script might get exited due to that. In order to update all the other containers correctly, you need to run this script with the same settings a second time.)
Waiting for watchtower to stop
Creating daily backup...
Waiting for backup container to stop
....
_....all other lines omitted for clarity_
....
Waiting for backup container to stop
Starting and updating containers...

Fatal error: Uncaught GuzzleHttp\Exception\ServerException: Server error: `POST http://localhost/v1.41/containers/nextcloud-aio-imaginary/start` resulted in a `500 Internal Server Error` response:
{"message":"failed to create task for container: failed to create shim task: OCI runtime create failed: runc create fail (truncated...)
 in /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113

ace:
#0 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Middleware.php(72): GuzzleHttp\Exception\RequestException::create(Object(GuzzleHttp\Psr7\Request), Object(GuzzleHttp\Psr7\Response), NULL, Array, NULL)
#1 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(209): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Response))
#2 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(158): GuzzleHttp\Promise\Promise::callHandler(1, Object(GuzzleHttp\Psr7\Response), NULL)
#3 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/TaskQueue.php(52): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}()
#4 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(251): GuzzleHttp\Promise\TaskQueue->run(true)
#5 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(227): GuzzleHttp\Promise\Promise->invokeWaitFn()
#6 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(272): GuzzleHttp\Promise\Promise->waitIfPending()
#7 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(229): GuzzleHttp\Promise\Promise->invokeWaitList()
#8 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(69): GuzzleHttp\Promise\Promise->waitIfPending()
#9 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Client.php(189): GuzzleHttp\Promise\Promise->wait()
#10 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/ClientTrait.php(95): GuzzleHttp\Client->request('POST', 'http://localhos...', Array)
#11 /var/www/docker-aio/php/src/Docker/DockerActionManager.php(189): GuzzleHttp\Client->post('http://localhos...')
#12 /var/www/docker-aio/php/src/Controller/DockerController.php(57): AIO\Docker\DockerActionManager->StartContainer(Object(AIO\Container\Container))
#13 /var/www/docker-aio/php/src/Controller/DockerController.php(33): AIO\Controller\DockerController->PerformRecursiveContainerStart('nextcloud-aio-i...', true)
#14 /var/www/docker-aio/php/src/Controller/DockerController.php(33): AIO\Controller\DockerController->PerformRecursiveContainerStart('nextcloud-aio-n...', true)
#15 /var/www/docker-aio/php/src/Controller/DockerController.php(202): AIO\Controller\DockerController->PerformRecursiveContainerStart('nextcloud-aio-a...', true)
#16 /var/www/docker-aio/php/src/Cron/StartAndUpdateContainers.php(20): AIO\Controller\DockerController->startTopContainer(true)
#17 {main}
  thrown in /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php on line 113
Something seems to be wrong: Nextcloud should be started at this step.
Sending backup notification...
Daily backup successful! Only logging successful backup and not sending backup notification since that has been disabled! You can get further info by looking at the backup logs in the AIO interface.
Daily backup script has finished
Total reclaimed space: 0B
++ head -1 /mnt/docker-aio-config/data/daily_backup_time
+ BACKUP_TIME=03:00
+ export BACKUP_TIME
+ export DAILY_BACKUP=1
+ DAILY_BACKUP=1
++ sed -n 2p /mnt/docker-aio-config/data/daily_backup_time
+ '[' '' '!=' automaticUpdatesAreNotEnabled ']'
+ export AUTOMATIC_UPDATES=1
+ AUTOMATIC_UPDATES=1
++ sed -n 3p /mnt/docker-aio-config/data/daily_backup_time
+ '[' successNotificationsAreNotEnabled '!=' successNotificationsAreNotEnabled ']'
+ export SEND_SUCCESS_NOTIFICATIONS=0
+ SEND_SUCCESS_NOTIFICATIONS=0
+ set +x
Deleting duplicate sessions
```

**Comment by szaimen** [comment]
Hi, what would be your suggestion? To check if docker hub is reachable before starting daily backups? Or do you have something different in mind?

**Comment by sunnyd24** [comment]
> Hi, what would be your suggestion? To check if docker hub is reachable before starting daily backups? Or do you have something different in mind?

@szaimen I think you may have already got this sorted as this is now closed, thanks for the speedy action. 
Docker hub check seems to be reasonable. 👍

Any thoughts on master container notifications service or something similar? 
Any of those I suggested earlier an option to implement, i.e. PushOver/PushBullet/Telegram/WebSocket?

**Comment by szaimen** [comment]
This is now released with v7.13.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
